Content.Server/Teleportation/TeleportLocationsSystem.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Content.Server.Chat.Systems;
+using Content.Server.Warps;
+using Content.Shared.Teleportation;
+using Content.Shared.Teleportation.Components;
+using Content.Shared.Teleportation.Systems;
+
+namespace Content.Server.Teleportation;
+
+/// <summary>
+/// <inheritdoc cref="SharedTeleportLocationsSystem"/>
+/// </summary>
+public sealed partial class TeleportLocationsSystem : SharedTeleportLocationsSystem
+{
+    [Dependency] private readonly ChatSystem _chat = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TeleportLocationsComponent, MapInitEvent>(OnInit);
+        SubscribeLocalEvent<TeleportLocationsComponent, TeleportLocationRequestPointsMessage>(OnTeleportLocationRequest);
+    }
+
+    private void OnInit(Entity<TeleportLocationsComponent> ent, ref MapInitEvent args)
+    {
+        UpdateTeleportPoints(ent);
+    }
+
+    private void OnTeleportLocationRequest(Entity<TeleportLocationsComponent> ent, ref TeleportLocationRequestPointsMessage args)
+    {
+        UpdateTeleportPoints(ent);
+    }
```

Comment:
Sloth made this comment already but instead of querying the server for this from the client, just call this function on a sub to `UiOpenedEvent` (you already do it in the shared code)